tests/test_methods_and_attributes.cpp

```diff
@@ -327,6 +327,18 @@ TEST_SUBMODULE(methods_and_attributes, m) {
     py::class_<MetaclassOverride>(m, "MetaclassOverride", py::metaclass((PyObject *) &PyType_Type))
         .def_property_readonly_static("readonly", [](py::object) { return 1; });
 
+    // test_overload_ordering
+    struct OverloadOrder {
+        int value;
+        OverloadOrder(int value) : value(value) {}
+    };
+    py::class_<OverloadOrder>(m, "OverloadOrder")
```

Comment:
Using a class here seems a little overkilled; it ought to be simpler to just overload an ordinary function:
```C++
m.def("overload_order", [](std::string) { return 1; });
m.def("overload_order", [](std::string) { return 2; });
m.def("overload_order", [](int) { return 3; });
m.def("overload_order", [](int) { return 4; }, py::prepend{});
```